include/deal.II/grid/grid_tools.h

```diff
@@ -1213,6 +1213,21 @@ namespace GridTools
     template <int dim, int spacedim>
     struct DistributedComputePointLocationsInternal
     {
+      DistributedComputePointLocationsInternal();
+
+      /**
+       * Function which sets up @p send_ranks, @p send_ptrs, @p recv_ranks,
+       * and @p recv_ptrs from @p send_components, @p recv_components,
+       * and @p n_searched_points
+       */
+      void
+      sort_components_and_init_crs_structures(const bool perform_handshake);
```

Comment:
```suggestion
      finalize_setup(const bool perform_handshake);
```

Reply:
I actually think we can remove the bool and internally check `recv_components.empy()==false`. Can't we?

Reply:
Yes this works :) I prefer to check if ```recv_components.size()>0``` in [3add97c](https://github.com/dealii/dealii/pull/15154/commits/3add97c063739a9fc0f996c37d2e1d772c4c7f64). Hope this is ok for you :)